doc/src/base/scopedvalues.md

```diff
@@ -0,0 +1,115 @@
+# Scoped Values
+
+Scoped values provide an implementation of dynamical scoping in Julia.
```

Comment:
i wonder if it would be more user-friendly to provide a very short description of what "dynamic scoping" is here, or at least link to one?

E.g. here's [wikipedia's](https://en.wikipedia.org/wiki/Scope_(computer_science)#Lexical_scope_vs._dynamic_scope_2)

> in dynamic scope (or dynamic scoping), if a variable name's scope is a certain function, then its scope is the time-period during which the function is executing: while the function is running, the variable name exists, and is bound to its value, but after the function returns, the variable name does not exist. This means that if function f invokes a separately defined function g, then under lexical scope, function g does not have access to f's local variables (assuming the text of g is not inside the text of f), while under dynamic scope, function g does have access to f's local variables (since g is invoked during the invocation of f). 

Maybe we could edit that down to helpful one line definition? I think these docs are really good, but i think currently the text assumes you know what dynamic scope is, while the examples are teaching you what it is.